src/Microsoft.ApiDesignGuidelines.Analyzers/Core/CollectionsShouldImplementGenericInterface.cs

```diff
@@ -4,36 +4,76 @@
         private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.CollectionsShouldImplementGenericInterfaceDescription), MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
-        
+
         internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
                                                                              s_localizableTitle,
                                                                              s_localizableMessage,
                                                                              DiagnosticCategory.Design,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182132.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.RegisterCompilationStartAction(
+               (context) =>
+               {
+                   var collectionType = WellKnownTypes.ICollection(context.Compilation);
+                   var genericCollectionType = WellKnownTypes.GenericICollection(context.Compilation);
+                   var enumerableType = WellKnownTypes.IEnumerable(context.Compilation);
+                   var genericEnumerableType = WellKnownTypes.GenericIEnumerable(context.Compilation);
+                   var listType = WellKnownTypes.IList(context.Compilation);
+                   var genericListType = WellKnownTypes.GenericIList(context.Compilation);
+
+                   if (collectionType == null && genericCollectionType ==null &&
+                       enumerableType == null && genericEnumerableType == null &&
+                       listType == null && genericListType == null )
+                   {
+                       return;
+                   }
+
+                   context.RegisterSymbolAction(c => AnalyzeSymbol(c, collectionType, genericCollectionType, enumerableType, genericEnumerableType, listType, genericListType), SymbolKind.NamedType);
```

Comment:
If you use a lambda, rather than a named method, for `AnalyzeSymbol`, then you won't have to pass any of the arguments because they'll be available through the closure.